### PR TITLE
[FX-4582] Migrate Modal to TailwindCSS

### PR DESCRIPTION
**Pull request by TomasSlama**
[FX-4582]

### Description

Migrate Modal and its components to TailwindCSS.

### How to test

- [Temploy](https://picasso.toptal.net/fx-4582-modal)
- open [story](https://picasso.toptal.net/?path=/story/overlays-modal--modal) on production and compare with [temploy](https://picasso.toptal.net/fx-4582-modal?path=/story/overlays-modal--modal). It should look the same.
- you can check SP https://github.com/toptal/staff-portal/pull/12331
- you can check CP https://github.com/toptal/client-portal/pull/8387

### Screenshots

SP
image
-----
CP
image

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

**Breaking change**

- codemod is created and showcased in the changeset
- [x] test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4582]: https://toptal-core.atlassian.net/browse/FX-4582?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by TomasSlama** [comment]
@toptal-bot run package:alpha-release

**Comment by TomasSlama** [comment]
@toptal-bot run package:alpha-release

**Review by dmaklygin — APPROVED** [review]
Compared temploys, looks equal

**Comment by TomasSlama** [comment]
@toptal-anvil ping reviewers

**Comment by TomasSlama** [comment]
@toptal-anvil ping reviewers

File: packages/base/Modal/src/Modal/Modal.tsx [hunk 1]
```diff
@@ -218,44 +216,55 @@ export const Modal = forwardRef<HTMLElement, Props>(function Modal(props, ref) {
...
+        'fixed z-modal inset-0 flex flex-col text-lg justify-center items-center'
+      )}
       style={style}
-      fullScreen={isExtraSmall}
+      slots={{
+        backdrop: Backdrop,
+      }}
+      closeAfterTransition
+      slotProps={{
+        // @ts-ignore
```
Comment (mkrl) [inline packages/base/Modal/src/Modal/Modal.tsx:235]: Do you think you can replace this with `// @ts-expect-error` with an explaining comment if it's not possible to fix this type error?

**Review by mkrl — APPROVED** [review]
Nice job, looks good! Left one comment, but it's not critical.

File: packages/base/Modal/src/Modal/Modal.tsx (hunk 1 above)
Comment (TomasSlama) [reply packages/base/Modal/src/Modal/Modal.tsx:235]: I will fix it 🙇

Comment (TomasSlama) [reply packages/base/Modal/src/Modal/Modal.tsx]: fixed 🙇

**Pull request merged**